### PR TITLE
Fixes `DataView` and `Promise` toString

### LINKED ISSUE #656: DataView does not have correct toStringTag
In Edge 14.14295 a data view has the incorrect [toStringTag](http://www.ecma-international.org/ecma-262/6.0/#sec-dataview.prototype-@@tostringtag).
```js
Object.prototype.toString.call(new DataView(new ArrayBuffer(23)));
// => [object Object] instead of [object DataView]
```

### PR DESCRIPTION
**Pull request by obastemur**
Fixes #656

### CODE REVIEW

**Comment by obastemur** [comment]
See https://github.com/Microsoft/ChakraCore/pull/657

**Comment by dilijev** [comment]
LGTM

**Comment by dilijev** [comment]
Hi @obastemur could you add a couple of unit tests for these cases which fail before your change and pass afterwards?

**Comment by dilijev** [comment]
@obastemur Also you'll need to rebase this on latest because the branch is out of date.

**Comment by obastemur** [comment]
@dilijev

> could you add a couple of unit tests for these cases which fail before your change and pass afterwards?

> Also you'll need to rebase this on latest because the branch is out of date.

Sure. PR is updated.

File: test/typedarray/samethread.baseline [hunk 1]
```diff
@@ -20999,6 +20999,7 @@ undefined
 undefined
 undefined
 10
+[object DataView]
```
Comment (dilijev) [inline test/typedarray/samethread.baseline]: Is there a corresponding change to samethread.js?

**Comment by dilijev** [comment]
Looks good. Thanks. I'll bring this in for testing today and merge it this afternoon.

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows arm_debug please
test Windows arm_release please
test Windows arm_test please
test Windows x64_debug please
test Windows x64_release please
test Windows x64_test please
test Windows x86_debug please
test Windows x86_release please
test Windows x86_test please

**Pull request merged**